lambdas/transcode/transcode/index.py

```diff
@@ -0,0 +1,51 @@
+"""
+Generate video previews for videos in S3.
+"""
+import pathlib
+import subprocess
+import tempfile
+
+from t4_lambda_shared.decorator import api, validate
+from t4_lambda_shared.utils import get_default_origins
+
+
+SCHEMA = {
+    'type': 'object',
+    'properties': {
+        'url': {
+            'type': 'string'
+        }
+    },
+    'required': ['url'],
+    'additionalProperties': False
+}
+
+FFMPEG = pathlib.Path('/opt/bin/ffmpeg')
+FFMPEG_TIME_LIMIT = 20
+DURATION = 5
+WIDTH = 320
+HEIGHT = 240
+
+
+@api(cors_origins=get_default_origins())
+@validate(SCHEMA)
+def lambda_handler(request):
+    """
+    Generate thumbnails for images in S3
+    """
+    url = request.args['url']
+
+    with tempfile.NamedTemporaryFile(suffix='.mp4') as output_file:
```

Comment:
Can we run out of disk here (over 512MB)? I guess the Lambda just fails at that point.